test/common/thread/lock_guard_test.cc

```diff
@@ -0,0 +1,27 @@
+#include "common/common/thread.h"
+
+#include "gtest/gtest.h"
+#include "library/common/thread/lock_guard.h"
+
+namespace Envoy {
+namespace Thread {
+
+class ThreadTest : public testing::Test {
+protected:
+  ThreadTest() = default;
+  int a_ ABSL_GUARDED_BY(a_mutex_){0};
+  MutexBasicLockable a_mutex_;
```

Comment:
Would it make more sense to create these properties inside the test case instead of within the thread class to remove state between the existing test case and potential future cases?

Reply:
each test starts with a fresh instance of the test class.

Reply:
Ah cool, TIL